src/contracts/GPv2VaultRelayer.sol

```diff
@@ -44,4 +44,91 @@ contract GPv2VaultRelayer {
     {
         vault.transferFromAccounts(msg.sender, transfers);
     }
+
+    /// @dev Performs a Balancer batched swap on behalf of a user and sends a
+    /// fee to the caller.
+    ///
+    /// This function reverts if:
+    /// - The caller is not the creator of the vault relayer
+    /// - The swap fails
+    /// - The fee transfer fails
```

Comment:
I'd also add that the transaction fails if this the deadline has passed (well, I assume so at least). This is because it took me a while to find out that `validTo` was still enforced in the fast path.
In general, I wonder if it makes sense to trigger a revert immediately when decoding an expired order instead of waiting until the settlement. I don't think we ever need to decode an expired order anywhere.

Reply:
I'll add a comment about how the validity is verified by the vault swap.

In general I'm not a huge fan of splitting up the the order validation, since, as it currently is, all the checks on whether or not the order is valid for a full settlement happens in the `computeTradeExecution` function (which I think is nice for legibility).

Reply:
> In general I'm not a huge fan of splitting up the the order validation

To me, the only reason why the validity is part of the order struct is because this struct is the same struct that is signed by the user to create an order. Otherwise, it could just be part of the trade, and then validity verification could be treated in the same way as signature verification. But I don't see any nice way to do this in a gas efficient way.

Reply:
In #526 I added some comments explaining how the order validity and limit price are being verified. Ultimately, we need to add E2E tests to check if this is actually the case.